compiler/src/org.graalvm.compiler.nodes/src/org/graalvm/compiler/nodes/java/MethodCallTargetNode.java

```diff
@@ -187,6 +187,11 @@ public void simplify(SimplifierTool tool) {
             // check if the type of the receiver can narrow the result
             ValueNode receiver = receiver();
 
+            ResolvedJavaType referencedReceiverType = referencedType();
```

Comment:
I wonder if we need/should drop the invokeKind().isInterface() condition on Line 185:

```
$ git diff
diff --git a/compiler/src/org.graalvm.compiler.nodes/src/org/graalvm/compiler/nodes/java/MethodCallTargetNode.java b/compiler/src/org.graalvm.compiler.nodes/src/org/graalvm/compiler/nodes/java/MethodCallTargetNode.java
index 1783483d77c..6dbf78ceccf 100644
--- a/compiler/src/org.graalvm.compiler.nodes/src/org/graalvm/compiler/nodes/java/MethodCallTargetNode.java
+++ b/compiler/src/org.graalvm.compiler.nodes/src/org/graalvm/compiler/nodes/java/MethodCallTargetNode.java
@@ -182,7 +182,7 @@ public class MethodCallTargetNode extends CallTargetNode implements IterableNode
          * Even though we are not registering an assumption (see comment below), the optimization is
          * only valid when speculative optimizations are enabled.
          */
-        if (invokeKind().isIndirect() && invokeKind().isInterface() && assumptions != null) {
+        if (invokeKind().isIndirect() && assumptions != null) {
 
             // check if the type of the receiver can narrow the result
             ValueNode receiver = receiver();
```

Reply:
I think we shouldn't remove it (it's not removed upstream) https://github.com/tkrodriguez/graal/blob/topic/GR-20002/compiler/src/org.graalvm.compiler.nodes/src/org/graalvm/compiler/nodes/java/MethodCallTargetNode.java#L181

I see `invokeKind().isIndirect()` missing upstream but I think this is related to a different patch that we should backport if need be.

Reply:
Right, my thinking was that this check be done for virtual invocations too. However, I'm not sure that's a possible path in this context. It passes the test so I'm OK for this to be merged.